src/buildout/wheel/__init__.py

```diff
@@ -1,56 +1,222 @@
+    """
+
+    def __init__(self, search_path, dest):
+        self.__dest = pkg_resources.normalize_path(dest)
+        pkg_resources.Environment.__init__(self, search_path)
 
-wheel_supported = pip.pep425tags.get_supported
+    def scan(self, search_path=None):
+        # replace `self.__dest` (the `eggs-directory`) inside `search_path`
+        # with all distributions (egg or otherwise) located in `self.__dest`
+        # since `pkg_resources.Environment` can detect eggs in `self.__dest`,
+        # but not other kinds of dists:
+        if (search_path and
+                pkg_resources.normalize_path(search_path[0]) == self.__dest):
+            search_path[0:1] = _get_dest_dist_paths(self.__dest)
+        return pkg_resources.Environment.scan(self, search_path)
 
-def interpret_distro_name(
-    location, basename, metadata, py_version=None,
-    precedence=pkg_resources.SOURCE_DIST,
-    platform=None
-    ):
-    if '.whl' in location:
-        path = parse.urlparse(location).path
+    def add(self, dist):
+        # pkg_resources.find_on_path(), used by self.scan() via
+        # find_distributions(), creates all non-.egg dists as
+        # precedence=DEVELOP_DIST.
+        # Fix precedence to EGG_DIST of all packages under `self.__dest`
+        if dist.location.startswith(self.__dest + '/'):
+            dist.precedence = pkg_resources.EGG_DIST
+        return pkg_resources.Environment.add(self, dist)
+
+
+class Installer(orig_Installer):
+    """
+    zc.buildout.easy_install.Installer class that feeds all dists in
+    `eggs-directory` to `pkg_resources.Environment`, not just the eggs that
+    `Environment` would auto-detect, and knows how to install `wheels`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        orig_Installer.__init__(self, *args, **kwargs)
+        # Sorry to throw away all the previous hard work creating self._env
+        # but we need to replace it with our special Environment.
+        self._env = Environment(self._path, dest=self._dest)
+
+    def _call_easy_install(self, spec, ws, dest, dist):
+        if not spec.endswith('.whl'):
+            return orig_Installer._call_easy_install(spec, ws, dest, dist)
+        tmp = tempfile.mkdtemp(dir=dest)
         try:
-            wf = wheel.install.WheelFile(path, context=wheel_supported)
+            # this causes the dist tree to be written twice in tempdirs inside
+            # `dest`, but the alternative would be to trick
+            # `setuptools.archive_util.unpack_archive` into unpacking wheels
+            # which requirs dealing with
+            # `setuptools.archive_util.unpack_archive.default_filter`.
+            dist = WheelDir(spec).install_into(tmp)
+            move = zc.buildout.easy_install._move_to_eggs_dir_and_compile
+            newloc = move(dist, dest)
+            return pkg_resources.Environment([newloc])[dist.project_name]
+        finally:
+            shutil.rmtree(tmp)
+
+
+class WheelDir(wheel.install.WheelFile):
```

Comment:
Sure, the only overridden method here was in `__init__`, and that was just to change the default value of a parameter.